### PR TITLE
Dev Env: Enable remote write for prometheus

### PR DESCRIPTION
**Pull request by toddtreece**
**What this PR does / why we need it**:

This adds `--enable-feature=remote-write-receiver` to the [default command from the prometheus Dockerfile](https://github.com/prometheus/prometheus/blob/edfe657b54057e427469a2151153d305270662a4/Dockerfile#L25-L28) in the prometheus docker compose file to make it easier to configure recorded queries in development.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: devenv/docker/blocks/prometheus/docker-compose.yaml [hunk 1]
```diff
@@ -4,6 +4,12 @@
       - "9090:9090"
     extra_hosts:
       - "host.docker.internal:host-gateway"
+    command: >
+      --enable-feature=remote-write-receiver
```
Comment (ifrost) [inline devenv/docker/blocks/prometheus/docker-compose.yaml:8]: I think this one is deprecated according to https://prometheus.io/docs/prometheus/latest/feature_flags/#remote-write-receiver

Could we change it to?

```suggestion
     --web.enable-remote-write-receiver
```

**Review by ifrost — CHANGES_REQUESTED** [review]

File: devenv/docker/blocks/prometheus/docker-compose.yaml (hunk 1 above)
Comment (toddtreece) [reply devenv/docker/blocks/prometheus/docker-compose.yaml:8]: @ifrost i noticed that as well, but it didn't work for me locally

Comment (ifrost) [reply devenv/docker/blocks/prometheus/docker-compose.yaml:8]: Hmm... seems to work on my machine :D I'll check it again

Comment (ifrost) [reply devenv/docker/blocks/prometheus/docker-compose.yaml:8]: @toddtreece I was able to add remote write target and record queries with `--web.enable-remote-write-receiver` 🤔 Are you getting any errors?

Comment (toddtreece) [reply devenv/docker/blocks/prometheus/docker-compose.yaml:8]: @ifrost i don't get errors when running `make devenv sources=prometheus`, but prometheus doesn't actually start for me.

here's the error i get:
```
$ make devenv sources=prometheus && docker logs devenv_prometheus_1
error parsing commandline arguments: unknown long flag '--web.enable-remote-write-receiver'
prometheus: error: unknown long flag '--web.enable-remote-write-receiver'
```

Comment (ifrost) [reply devenv/docker/blocks/prometheus/docker-compose.yaml:8]: Hmm... fair enough 🤔 Not a big deal, we will get an error once it's removed anyway 👍

**Review by ifrost — APPROVED** [review]

**Pull request merged**